packages/liquidity-monitor-backend/src/provider-monitoring/provider-monitoring.service.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable, Logger, OnModuleDestroy, OnModuleInit } from '@nestjs/common';
+import { InjectDataSource } from "@nestjs/typeorm";
+import { DataSource } from "typeorm";
+import { Notifications, TelegrafService } from "@boarteam/boar-pack-users-backend";
+import { fmt, bold } from "telegraf/format";
+
+type TProviderActivity = {
+  latestQuoteDate: string,
+  name: string,
+  threshold: number
+}
+
+@Injectable()
+export class ProviderMonitoringService implements OnModuleInit, OnModuleDestroy {
+  private readonly logger = new Logger(ProviderMonitoringService.name);
+  // We'll store here notification attempts and retry attempts with exponential backoff (seconds)
+  // { providerName: { attempts: 0 - 4, lastNotification: Date }}
+  private notificationsJournal = new Map<string, {
+    attempts: number,
+    lastNotification: Date
+  }>();
+  private readonly exponentialBackoff = [299, 599, 1799, 3599];
+  private providerActivityInterval: NodeJS.Timeout | null = null;
+
+  constructor(
+    @InjectDataSource()
+    private readonly dataSource: DataSource,
+    private readonly telegrafService: TelegrafService,
+  ) {
+  }
+
+  async onModuleInit() {
+    const setting = await this.getSetting();
+    if (setting === 'yes') {
+      this.startMonitoring();
+    }
+  }
+
+  onModuleDestroy() {
+    this.stopMonitoring();
+  }
+
+  private async getSetting(): Promise<string> {
+    const result = await this.dataSource.query(`
+      select value from settings where key = '${Notifications.QuotesByProviderStatus}'
+    `);
+    return result?.[0]?.value || 'no';
+  }
```

Comment:
that should live inside settings module
```suggestion
    const result = await this.settings.getValue(Notifications.QuotesByProviderStatus);
    return result || 'no';
  }
```

Reply:
Replaced to `settingsService.getSettings` method